web/admin/src/App.js

```diff
@@ -266,29 +267,54 @@ class App extends PureComponent {
   }
 
   doPublish = content => {
+
     // 1. Remove all derived copies of all content
     this.publicContentRef().remove()
     this.usersRef().remove()
     this.tiersRef().remove()
 
     // 2. Create derived copies from `publishedContent`
     const { key, ...contentToPublish } = content
+    let csvData = []
+
+    //check if new object is for csv
+    if (content.rawData) {
+      const publishData = content.rawData.slice()
+
+      //add key to data then remove so not published publically
+      csvData = this.publishCSVData(publishData, key)
+      delete content.rawData
+    }
+  
     const publishedContent = Object.keys({...this.state.publishedContent, [key]: content})
     .map(k => k === key ? content : {...this.state.publishedContent[k], key: k})
     .filter(x => Object.keys(x).length > 1) // Ignore key-only objects that are being unpublished
-    
+
+    //check for reordering thus content blank
+    if (Object.keys(content).length === 0) {
+      publishedContent.forEach(item => {
+        if (item.type === "textCSV" || item.type === "webCSV" || item.type === "videoCSV") {
+          let newOrderData = item.rawData.slice()
```

Comment:
Why `slice()`?  Is it necessary?  It just creates a new array that has all the same elements.

Reply:
I'm assuming its either legacy code or ensuring a object to not change the underlying info. Let me change to object.assign

Reply:
Yes this is to make a new array without changing the underlying value